components/eam/src/physics/cam/nucleate_ice_cam.F90

```diff
@@ -344,6 +356,39 @@ subroutine nucleate_ice_cam_init(mincld_in, bulk_scale_in)
             coarse_dust_idx, coarse_nacl_idx
          call endrun(routine//': ERROR required mode-species type not found')
       end if
+      if (mode_strat_coarse_idx>0) then
+         call rad_cnst_get_info(0, mode_strat_coarse_idx, nspec=nspec)
+         do n = 1, nspec
+            call rad_cnst_get_info(0, mode_strat_coarse_idx, n, spec_type=str32)
+            select case (trim(str32))
+            case ('sulfate')
+               strat_coarse_so4_idx = n    ! find sulfate
+            end select
+         end do
+      endif
+      !for accumulation mode
+      if (mode_accum_idx>0) then
+         call rad_cnst_get_info(0, mode_accum_idx, nspec=nspec)
+         do n = 1, nspec
+            call rad_cnst_get_info(0, mode_accum_idx, n, spec_type=str32)
+            select case (trim(str32))
+            case ('sulfate')
+               accum_so4_idx = n    ! find sulfate
+            case ('p-organic')
+               accum_pom_idx = n    ! find pom
+            case ('s-organic')
+               accum_soa_idx = n    ! find soa
+            case ('black-c')
+               accum_bc_idx = n     ! find bc
+            case ('dust')
+               accum_dst_idx = n    ! find dust
+            case ('seasalt')
+               accum_ncl_idx = n    ! find dust 
+            case ('m-organic')
+               accum_mom_idx = n    ! find dust   
+            end select
+         end do
+      endif
```

Comment:
Do we need those accum variables?

Reply:
I want to keep this part. It is useful to add in better ice nucleation for the future and it doesn't consume much computational resource.